src/layouts/SiteLaunchPad/components/SiteLaunchCheckList.tsx

```diff
@@ -12,11 +37,379 @@ export const SiteLaunchChecklistTitle = (): JSX.Element => {
+                <Text>{record.type}</Text>
+              </Td>
+              <Td textStyle="body-2" color="text.label.secondary">
+                {textWithCopyIcon(record.target)}
+              </Td>
+            </Tr>
+          ))}
+        </Tbody>
+      </Table>
+    </TableContainer>
+  )
+}
+
+interface SiteLaunchChecklistBodyProps {
+  setPageNumber: (number: number) => void
+  handleIncrementStepNumber: () => void
+  handleDecrementStepNumber: () => void
+}
+export const SiteLaunchChecklistBody = ({
+  setPageNumber,
+  handleIncrementStepNumber,
+  handleDecrementStepNumber,
+}: SiteLaunchChecklistBodyProps): JSX.Element => {
   const {
     siteLaunchStatusProps,
     setSiteLaunchStatusProps,
   } = useSiteLaunchContext()
-  // todo site launch pad checklist table
-  return <SiteLaunchPadBody />
+
+  // todo remove this mock after IS-219
+  const mockDNSRecords = () => {
+    if (siteLaunchStatusProps) {
+      setTimeout(() => {
+        setSiteLaunchStatusProps({
+          stepNumber: SITE_LAUNCH_TASKS_LENGTH,
+          siteLaunchStatus: "LAUNCHING",
+          dnsRecords: [
+            {
+              source: "www.isomer.gov.sg",
+              type: "CNAME",
+              target: "dodsfdsag34fd2.cloudfront.net",
+            },
+            {
+              source: "isomer.gov.sg",
+              type: "A",
+              target: "18.1.2.3",
+            },
+            {
+              source: "_ba2bfdc9cd3388ff4427040b865afacf5.isomer.gov.sg",
+              type: "CNAME",
+              target:
+                "_617497b1eed3650c89eb59582d89d085.xmjnffzjyj.acm-validations.aws",
+            },
+          ],
+        })
+      }, 10000)
+    }
+  }
+
+  const handleGenerateDNSRecordsOnClick = () => {
+    setTasksDone(tasksDone + 1)
+    setSiteLaunchStatusProps({
+      ...siteLaunchStatusProps,
+      stepNumber: SITE_LAUNCH_TASKS.GENERATE_NEW_DNS_RECORDS,
+      siteLaunchStatus: "LAUNCHING",
+    })
+    // todo: actually generate DNS records
+    mockDNSRecords()
+  }
+
+  const [tasksDone, setTasksDone] = useState<number>(
+    siteLaunchStatusProps?.stepNumber ?? SITE_LAUNCH_TASKS.NOT_STARTED
+  )
+  const checkboxes = []
+  const numberOfTasks = SITE_LAUNCH_TASKS_LENGTH
+  const numberOfCheckboxes = numberOfTasks - 1 // last task is a button
+
+  for (let i = 0; i < numberOfCheckboxes; i += 1) {
+    checkboxes.push(
+      <Center>
+        <Checkbox
+          // we want use to check box in order, so we disable it when it is not the next one
+          isDisabled={tasksDone >= i + 2 || tasksDone < i}
+          isChecked={i < tasksDone}
+          onChange={(e) => {
+            e.preventDefault()
+            if (e.target.checked) {
+              setTasksDone(i + 1)
+              handleIncrementStepNumber()
+            } else {
+              setTasksDone(i)
+              handleDecrementStepNumber()
+            }
+          }}
+        />
+      </Center>
+    )
+  }
```

Comment:
sure, c19fbd1853fe427bcb33220c7fb3c8e843e4e826
but could I get you help to explain why `map` is preferred? there is pretty much no code change code within the block no?

Reply:
technically no, but `map` is idiomatic in react for a few reasons:

1. `map` always regenerates the array. because react uses object ref to re-render, doing a push means that it might lead to the array not re-rendering
2. `map` is a (supposedly) pure function, which means that there are no side effects. hence, using them for arrays is a hint that the only thing is a transformation from some data into a jsx component.

also, from [react docs](https://react.dev/learn/keeping-components-pure#local-mutation-your-components-little-secret)

> Writing pure functions takes some habit and discipline. But it also unlocks marvelous opportunities:
>
> Your components could run in a different environment—for example, on the server! Since they return the same result for the same inputs, one component can serve many user requests.
> You can improve performance by [skipping rendering](https://react.dev/reference/react/memo) components whose inputs have not changed. This is safe because pure functions always return the same results, so they are safe to cache.
>
> If some data changes in the middle of rendering a deep component tree, React can restart rendering without wasting time to finish the outdated render. Purity makes it safe to stop calculating at any time.
>
> Every new React feature we’re building takes advantage of purity. From data fetching to animations to performance, keeping components pure unlocks the power of the React paradigm.